backend/sirius-components-diagrams-layout/src/test/java/org/eclipse/sirius/components/diagrams/layout/incremental/NodeCreationTests.java

```diff
@@ -137,20 +137,89 @@ public void testNodeCreationOutsideItsParent() {
-        assertEquals(bNode.getSize().getWidth(), 200, epsilon);
-        assertEquals(bNode.getSize().getHeight(), 300, epsilon);
+        assertThat(bNode.getPosition()).isEqualTo(Position.at(500, 500));
+        assertThat(bNode.getSize()).isEqualTo(Size.of(200, 300));
+    }
+
+    @Test
+    public void testChildlessNodeCreation() {
+        Path path = Paths.get("src", "test", "resources", "editing-contexts", "testChildlessNodeCreation"); //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$//$NON-NLS-4$ //$NON-NLS-5$
+        JsonBasedEditingContext editingContext = new JsonBasedEditingContext(path);
+
+        Position eventCreationPosition = Position.at(100, 100);
+        Diagram diagram = TestLayoutDiagramBuilder.diagram("Root").nodes().and().build(); //$NON-NLS-1$
+
+        Diagram layoutedDiagram = this.createNewNode(diagram, editingContext, eventCreationPosition);
+
+        // parent node should still be positioned at the eventCreationPosition
+        Optional<Node> optionalParent = this.getNode(layoutedDiagram.getNodes(), "Parent"); //$NON-NLS-1$
+        assertThat(optionalParent).isPresent();
+        Node parent = optionalParent.get();
+        Position parentPosition = parent.getPosition();
+        assertThat(parentPosition).isEqualTo(Position.at(100, 100));
+    }
+
+    @Test
+    public void testMultiNodeCreation() {
+        Path path = Paths.get("src", "test", "resources", "editing-contexts", "testNodeCreation"); //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$//$NON-NLS-4$ //$NON-NLS-5$
+        JsonBasedEditingContext editingContext = new JsonBasedEditingContext(path);
+
+        Position eventCreationPosition = Position.at(100, 100);
+        Diagram diagram = TestLayoutDiagramBuilder.diagram("Root").nodes().and().build(); //$NON-NLS-1$
+
+        Diagram layoutedDiagram = this.createNewNode(diagram, editingContext, eventCreationPosition);
+
+        // parent node should still be positioned at the eventCreationPosition
+        Optional<Node> optionalParent = this.getNode(layoutedDiagram.getNodes(), "Parent"); //$NON-NLS-1$
+        assertThat(optionalParent).isPresent();
+        Node parent = optionalParent.get();
+        Position parentPosition = parent.getPosition();
+        assertThat(parentPosition).isEqualTo(Position.at(100, 100));
+
+        // since the parent was not positioned yet, its children should be at a default position
+        Optional<Node> optionalChild = this.getNode(layoutedDiagram.getNodes(), "Child"); //$NON-NLS-1$
+        assertThat(optionalChild).isPresent();
+        Node child = optionalChild.get();
+        assertThat(child.getPosition()).isEqualTo(this.getDefaultPosition(parent, child));
+    }
+
+    /**
+     * The is the exact same method than {@link NodePositionProvider} except that it is for {@link Node}.
+     *
+     * @param node
+     *            The node
+     * @return its default position
+     */
+    private Position getDefaultPosition(Node parent, Node child) {
```

Comment:
Copy pasting this algorithm just to change the type of the parameter is not very elegant, but I did not have a better idea.